### ISSUE TITLE
Issue spawning yarn child-process on Windows

### ISSUE THREAD

**Issue by aramperes** [darionco/yarn-add-no-save#3]
Trying to run `yarn-add-no-save@1.0.2` on Windows will cause the following error:

```
(node:10020) UnhandledPromiseRejectionWarning: Error: spawn yarn ENOENT
    at _errnoException (util.js:992:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:190:19)
    at onErrorNT (internal/child_process.js:372:16)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)
(node:10020) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:10020) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
Done in 0.25s.
```

The actual error is `Error: spawn yarn ENOENT`, meaning Windows can't find the `yarn` executable during `child_process#spawn`. This can be demonstrated by using `node -p "require('child_process').spawn('yarn')"`.

Now the root cause is that even if Yarn is installed, it's actual name will always be `yarn.cmd`. Windows simply aliases `yarn` to `yarn.cmd` in the command prompt; however, this aliasing doesn't work when spawning child processes.

I can see two possible avenues for solving this:

1. Resolve the `yarn` command correctly, in a cross-platform manner using a library such as [`node-which`](https://www.npmjs.com/package/which). `require('which').sync('yarn'); // Returns C:\Path\To\yarn.CMD`

2. Figure out the path of `yarn` using an environment variable (from running a script). In a package.json script, you can use `NPM_EXECPATH` to reconstruct the yarn command. This won't work in a global context because it's not running a package.json script.

@darionco Would you accept a PR including `node-which`?

**Comment by darionco** [comment]
I would like to keep this package as lean as possible, `node-which` only depends on `isexe` which has no dependencies, so I think it is a good candidate to solve this issue and port this utility to windows.

I'd appreciate a PR, thank you

**Comment by aramperes** [comment]
Do you have a preference over using `npm` or `yarn` for locking, and `dependencies` or `devDependencies` for `node-which`?

**Comment by darionco** [comment]
since this is a yarn utility `yarn` would be better and since `node-which` will be used at runtime `dependencies`

**Comment by darionco** [comment]
Fix released in 1.0.3
Thank you for your help!

**Comment by aramperes** [comment]
No problem! I can now confirm that `yarn-add-no-save` works well on Windows :+1:
